pkg/storage/errors.go

```diff
@@ -96,15 +114,24 @@ func IsUnreachable(err error) bool {
 
 // IsTestFailed returns true if and only if err is a write conflict.
 func IsTestFailed(err error) bool {
-	return isErrCode(err, ErrCodeResourceVersionConflicts)
+	return isErrCode(err, ErrCodeResourceVersionConflicts, ErrCodeInvalidUID)
```

Comment:
Note that etcdHelper uses `ErrCodeResourceVersionConflicts` to indicate invalid UID. It'd better use a different error code. I would want to discuss it with @caesarxuchao and fix that in etcdHelper too. But that could be done in parallel and shouldn't block this PR. 

/cc @wojtek-t @xiang90